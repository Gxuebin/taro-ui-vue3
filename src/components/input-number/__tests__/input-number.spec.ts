import { mount } from '@vue/test-utils'
import AtInputNumber from '../index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtInputNumber as any, {
    slots,
    props: { ...values },
  })
}

describe('AtInputNumber', () => {
  it('should render default AtInputNumber', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- class', () => {
    const wrapper = factory({ class: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- type(number)', () => {
    const wrapper = factory({ type: 'number' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- type(digit)', () => {
    const wrapper = factory({ type: 'digit' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- disabled', () => {
    const wrapper = factory({ disabled: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- disabledInput', () => {
    const wrapper = factory({ disabledInput: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- value', () => {
    const wrapper = factory({ value: 2 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- width', () => {
    const wrapper = factory({ width: 200 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- size', () => {
    const wrapper = factory({ size: 'large' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- step', () => {
    const wrapper = factory({ step: 2 })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtNumberInput Behavior', () => {
  it('should trigger onChange', () => {
    const onChange = jest.fn()
    const wrapper = factory({ value: 2, onChange: onChange })
    wrapper.find('.at-input-number .at-input-number__input').trigger('input')
    expect(onChange).toBeCalled()
  })

  it('should trigger onBlur', () => {
    const onBlur = jest.fn()
    const wrapper = factory({ value: 2, onBlur: onBlur })
    wrapper.find('.at-input-number .at-input-number__input').trigger('blur')
    expect(onBlur).toBeCalled()
  })
})
