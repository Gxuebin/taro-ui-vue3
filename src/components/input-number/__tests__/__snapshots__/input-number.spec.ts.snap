// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtInputNumber should render default AtInputNumber 1`] = `
<view
  class="at-input-number"
>
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    style="width: 2.56rem;"
    type="number"
  />
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;

exports[`AtInputNumber should render prop -- class 1`] = `
<view
  class="test at-input-number test"
>
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    style="width: 2.56rem;"
    type="number"
  />
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;

exports[`AtInputNumber should render prop -- disabled 1`] = `
<view
  class="at-input-number"
>
  <view
    class="at-input-number__btn at-input-number--disabled"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    disabled=""
    style="width: 2.56rem;"
    type="number"
  />
  <view
    class="at-input-number__btn at-input-number--disabled"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;

exports[`AtInputNumber should render prop -- disabledInput 1`] = `
<view
  class="at-input-number"
>
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    disabled=""
    style="width: 2.56rem;"
    type="number"
  />
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;

exports[`AtInputNumber should render prop -- size 1`] = `
<view
  class="at-input-number at-input-number--lg"
>
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    style="width: 2.56rem;"
    type="number"
  />
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;

exports[`AtInputNumber should render prop -- step 1`] = `
<view
  class="at-input-number"
>
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    style="width: 2.56rem;"
    type="number"
  />
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;

exports[`AtInputNumber should render prop -- style 1`] = `
<view
  class="at-input-number"
>
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    style="width: 2.56rem;"
    type="number"
  />
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;

exports[`AtInputNumber should render prop -- type(digit) 1`] = `
<view
  class="at-input-number"
>
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    style="width: 2.56rem;"
    type="digit"
  />
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;

exports[`AtInputNumber should render prop -- type(number) 1`] = `
<view
  class="at-input-number"
>
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    style="width: 2.56rem;"
    type="number"
  />
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;

exports[`AtInputNumber should render prop -- value 1`] = `
<view
  class="at-input-number"
>
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    style="width: 2.56rem;"
    type="number"
  />
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;

exports[`AtInputNumber should render prop -- width 1`] = `
<view
  class="at-input-number"
>
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-subtract at-input-number__btn-subtract"
    />
  </view>
  <input
    class="at-input-number__input"
    style="width: 4.2667rem;"
    type="number"
  />
  <view
    class="at-input-number__btn"
  >
    <text
      class="at-icon at-icon-add at-input-number__btn-add"
    />
  </view>
</view>
`;
